Source/WTF/wtf/CompletionHandler.h

```diff
@@ -78,11 +78,27 @@ class CompletionHandler<Out(In...)> {
         return std::exchange(m_function, nullptr)(std::forward<In>(in)...);
     }
 
+    ThreadLikeAssertion callThread() const { return m_callThread; }
+
 private:
     Function<Out(In...)> m_function;
     NO_UNIQUE_ADDRESS ThreadLikeAssertion m_callThread;
 };
 
+template <typename CallableType>
+concept HasThreadAssertion = requires(CallableType callable)
+{
+    { callable.callThread() } -> std::same_as<ThreadLikeAssertion>;
+};
+
+template <typename CallableType>
+std::optional<ThreadLikeAssertion> threadAssertionFromCallable(const CallableType& callable)
```

Comment:
I don't know if there's use for resolving "the type did not have threadlike assertion"
So you might aswell return ThreadLikeAssertion directly.